app/javascript/components/pages/SightEdit.js

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from "react";
+import { Redirect } from "react-router-dom";
+import { Button, Form, FormGroup, Input, Label } from "reactstrap";
+
+export class SightEdit extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      form: {
+        name: this.props.sight.name,
+        address1: this.props.sight.address1,
+        city: this.props.sight.city,
+        zip_code: this.props.sight.zip_code,
+        state: this.props.sight.state,
+        country: this.props.sight.country,
+        phone: this.props.sight.phone,
+        trip_id: this.props.trip && this.props.trip.id,
+      },
```

Comment:
when the keys match properties, you can destructure the whole sight object from props